OPEN_PROBLEMS/HUMAN_READABLE_NAMING.md

```diff
@@ -3,13 +3,47 @@
+
 ## Long Description
 
-Memorizing Content Addresses is hard, No one should have to do that. We need a solution that can provide Human Readable Names to users without making them to rely on a central provider, an unauthenticated host, on unauthenticated data or without leaking their intent to access some piece of content that is referenced by that name.
+Content naming is an integral part of any content-oriented network architecture. In contrast to today’s approach of securing the channel and authenticating the end-host when receiving some content, in content-centric networks, it is the content itself that has to be secured and authenticated. Doing so enables the very important property of location-independent caching of content - that is, clients don’t need to trust the machine that provides the content, as long as one can verify that the content is what it claims to be. The claim is verified through the content name.
+
+That said, the naming scheme needs to integrate some properties to guarantee security and some others to enable non-security-related functionality.
+
+There are generally three types of names that have so far been investigated and implemented in the area of content-oriented networks: flat, normally hash-based names (as seen in IPFS and [NetInf](https://core.ac.uk/download/pdf/11433344.pdf)), hierarchical, as seen in [NDN](https://named-data.net) and [tag-based](http://conferences2.sigcomm.org/acm-icn/2014/papers/p17.pdf).
+
+We start by dividing between two types of properties for content names, that is, those that a name “has to have” and those that a name “would be good to have”.
+
+In a content-oriented architecture (such as IPFS) the name **has to:**
+
+- **provide security** guarantees, which further splits into two sub-properties
+  - be *persistent & unique:* the name needs to stay the same even if the storage location or the owner of some content changes. Unique names enable the very important property of being able to cache content in arbitrary points in the network. Even if a tiny portion of the content changes, the name identifier changes too. In IPFS, this is achieved with hashing. Name-persistency makes it difficult to address dynamic content - see discussion on *“mutable content”* further below.
+  - be *self-certifying:* the main purpose of having self-certifying names is to prove _data authenticity_. This is an important complementary part of name persistency to enable location-agnostic caching. Self-certifying names are normally non-human readable. 
+Traditionally, self-certifying names have been consider to be those that are not human readable and most commonly have been based on the hash of the content itself. Alternative ways to authenticate content is through obtaining the hash of the owner’s key. This helps in case of dynamic content and is used in IPNS and NDN, but it might lead to limited decentralisation properties.
```

Comment:
```suggestion
Traditionally, self-certifying names have been considered to be those that are not human-readable and most commonly have been based on the hash of the content itself. An alternative way to authenticate content is through the hash of the owner’s key. This helps in case of dynamic content and is used in IPNS and NDN, but it might lead to limited decentralisation properties.
```